test/run-drun/self-send-trap.mo

```diff
@@ -0,0 +1,294 @@
+//MOC-FLAG --trap-on-call-error
+//MOC-ENV MOC_UNLOCK_PRIM=yesplease
+import Prim "mo:⛔";
+
+actor self {
+
+  func showError(e : Error) : Text = debug_show (Prim.errorCode(e), Prim.errorMessage(e));
+
+  let MAX_SELF_QUEUE_CAPACITY = 500;
+  let PRED_MAX_SELF_QUEUE_CAPACITY = MAX_SELF_QUEUE_CAPACITY - 1 : Nat;
+  let DOUBLE_CAPACITY = 2 * MAX_SELF_QUEUE_CAPACITY;
+
+  let raw_rand = (actor "aaaaa-aa" : actor { raw_rand : () -> async Blob }).raw_rand;
+
+  public func request() : async () {
+  };
+
+  public func oneway() : () {
+  };
+
+  public func test1() : async () {
+    var n = 0;
+    while (n < DOUBLE_CAPACITY) {
+//      Prim.debugPrint(debug_show n);
```

Comment:
should these stay in this file (there are a bunch of them)

Reply:
Done